testingutils/src/test/java/com/github/awsjavakit/testingutils/networking/WiremockDirectCallClientTest.java

```diff
@@ -0,0 +1,263 @@
+import static com.github.awsjavakit.apigateway.HttpMethod.PUT;
+import static com.github.awsjavakit.testingutils.RandomDataGenerator.randomElement;
+import static com.github.awsjavakit.testingutils.RandomDataGenerator.randomInteger;
+import static com.github.awsjavakit.testingutils.RandomDataGenerator.randomString;
+import static com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder.like;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static java.util.Objects.nonNull;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.github.awsjavakit.apigateway.HttpMethod;
+import com.github.awsjavakit.misc.StringUtils;
+import com.github.awsjavakit.misc.paths.UriWrapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.direct.DirectCallHttpServerFactory;
+import com.github.tomakehurst.wiremock.http.HttpHeader;
+import com.github.tomakehurst.wiremock.http.ResponseDefinition;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiPredicate;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class WiremockDirectCallClientTest {
+
+  public static final String LOCALHOST = "https://localhost";
+  public static final BiPredicate<String, String> ALLOW_ALL_HEADERS = (header, value) -> true;
+  public static final String HEADER_DELIMITER = ",";
+  public static final String EMPTY = null;
+  private HttpClient directCallClient;
+  private WireMockServer directCallServer;
+
+  public static URI uriWithPath(String hostUri) {
+    return UriWrapper.fromUri(hostUri).addChild("some/path").getUri();
+  }
+
+  @BeforeEach
+  public void init() {
+    var factory = new DirectCallHttpServerFactory();
+    this.directCallServer = new WireMockServer(options().httpServerFactory(factory));
+    directCallServer.start(); // no-op, not required
+    var directCallHttpServer = factory.getHttpServer();
+    directCallClient = new WiremockDirectCallClient(directCallHttpServer);
+  }
+
+  @AfterEach
+  public void stop() {
+    this.directCallServer.stop();
```

Comment:
Remove `this`. If starting is not required, stopping is equally not required? (I am unsure about the comment on line 60)